multiuser/api/che-multiuser-api-authentication-commons/src/main/java/org/eclipse/che/multiuser/api/authentication/commons/SessionStore.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.multiuser.api.authentication.commons;
+
+import java.util.concurrent.ConcurrentHashMap;
+import javax.inject.Singleton;
+import javax.servlet.http.HttpSession;
+
+@Singleton
+public class SessionStore {
```

Comment:
Could you please write javadoc to this class, explaining how it should be used and what level of thread-safety it provides.

Reply:
also please test this class